src/components/notifications-table/NotificationsTable.jsx

```diff
@@ -0,0 +1,212 @@
+
+        return (
+            <ul>
+                {namesToDisplay.map(name => (
+                    <li key={name}>{name}</li>
+                ))}
+            </ul>
+        );
+    };
+
+    // TODO: We should fix d2-ui-components instead
+    getValueForUrl = value => {
+        return (
+            <a
+                rel="noopener noreferrer"
+                style={{ wordBreak: "break-all" }}
+                href={value}
+                target="_blank"
+            >
+                {value}
+            </a>
+        );
+    };
+
+    getMetadataTypes = notification => {
+        return this.getValueForCollection(notification.selectedTypes.map(type => ({ name: type })));
+    };
+
+    getAPILink = notification => {
+        const ids = _(notification.metadata)
+            .values()
+            .flatten()
+            .map(e => e.id)
+            .value();
+        return this.getValueForUrl(createMetadataExportUrl(this.props.d2, ids));
+    };
+
+    columns = [
+        { name: "user", text: i18n.t("User"), sortable: true },
+        { name: "timestamp", text: i18n.t("Timestamp"), sortable: true },
+        { name: "status", text: i18n.t("Status"), sortable: true },
+    ];
+
+    initialSorting = ["id", "asc"];
+
+    detailsFields = [
+        { name: "user", text: i18n.t("User") },
+        { name: "timestamp", text: i18n.t("Timestamp") },
+        { name: "status", text: i18n.t("Status") },
+        {
+            name: "metadata",
+            text: i18n.t("Metadata Types"),
+            getValue: notification => this.getMetadataTypes(notification),
+        },
+        {
+            name: "href",
+            text: i18n.t("API Link"),
+            getValue: notification => this.getAPILink(notification),
+        },
+    ];
+
+    actions = [
+        {
+            name: "details",
+            text: i18n.t("Details"),
+            multiple: false,
+            type: "details",
+        },
+        {
+            name: "delete",
+            text: i18n.t("Delete"),
+            multiple: false,
+            onClick: this.deleteNotification,
+        },
+        {
+            name: "summary",
+            text: i18n.t("View summary"),
+            icon: "description",
+            multiple: false,
+            onClick: this.openSummary,
+        },
+    ];
+
+    render() {
+        const { tableKey, toDelete, syncReport, summaryOpen } = this.state;
+        const { d2, classes } = this.props;
+
+        return (
+            <React.Fragment>
+                <ConfirmationDialog
+                    isOpen={!!toDelete}
+                    onSave={this.confirmDelete}
+                    onCancel={this.cancelDelete}
+                    title={i18n.t("Delete Instance?")}
+                    description={i18n.t("Are you sure you want to delete this instance?")}
+                    saveText={i18n.t("Ok")}
+                />
+                <PageHeader title={i18n.t("Notifications")} onBackClick={this.backHome} />
```

Comment:
[minor and subjective] Do you like leaving a blank line between components at the same level?